arangod/GeneralServer/AcceptorTcp.cpp

```diff
@@ -265,14 +273,22 @@ void AcceptorTcp<SocketType::Ssl>::asyncAccept() {
   // select the io context for this socket
   auto& ctx = _server.selectIoContext();
 
-  _asioSocket = std::make_unique<AsioSocket<SocketType::Ssl>>(ctx, _server.sslContexts());
+  {
+    std::unique_lock<std::mutex> guard(_mutex);
+    _asioSocket = std::make_unique<AsioSocket<SocketType::Ssl>>(ctx, _server.sslContexts());
+  }
   auto handler = [this](asio_ns::error_code const& ec) {
     if (ec) {
       handleError(ec);
       return;
     }
 
-    performHandshake(std::move(_asioSocket));
+    decltype(_asioSocket) as;
+    {
+      std::unique_lock<std::mutex> guard(_mutex);
+      as = std::move(_asioSocket);
+    }
+    performHandshake(std::move(as));
```

Comment:
We should check here if `as` is a nullptr. This could happen IMO if some other thread calls `close()` between the first few lines of the handler and line 286.